src/Nox.Generator/Application/Commands/DeleteByIdCommand.template.cs

```diff
@@ -38,29 +45,33 @@ public virtual async Task<bool> Handle(Delete{{entity.Name}}ByIdCommand request,
 	{
 		cancellationToken.ThrowIfCancellationRequested();
 		await OnExecutingAsync(request);
+		
+		foreach(var keyDto in request.KeyDtos)
+		{
+			{{- for key in entity.Keys }}
+			{{- keyType = SingleTypeForKey key }}
+			var key{{key.Name}} = {{codeGeneratorState.DomainNameSpace}}.{{entity.Name}}Metadata.Create{{key.Name}}(keyDto.key{{key.Name}});
+			{{- end }}		
 
-		{{- for key in entity.Keys }}
-		{{- keyType = SingleTypeForKey key }}
-		var key{{key.Name}} = {{codeGeneratorState.DomainNameSpace}}.{{entity.Name}}Metadata.Create{{key.Name}}(request.key{{key.Name}});
-		{{- end }}
+			var entity = await DbContext.{{entity.PluralName}}.FindAsync({{entity.Keys | array.map "Name" | keysQuery}});
+			if (entity == null{{if (entity.Persistence?.IsAudited ?? true)}} || entity.IsDeleted == true{{end}})
+			{
+				return false;
+			}
+			{{- if !entity.IsOwnedEntity }}
 
-		var entity = await DbContext.{{entity.PluralName}}.FindAsync({{primaryKeysQuery}});
-		if (entity == null{{if (entity.Persistence?.IsAudited ?? true)}} || entity.IsDeleted == true{{end}})
-		{
-			return false;
-		}
-		{{- if !entity.IsOwnedEntity }}
+			entity.Etag = request.Etag.HasValue ? request.Etag.Value : System.Guid.Empty;
+			{{- end }}
 
-		entity.Etag = request.Etag.HasValue ? request.Etag.Value : System.Guid.Empty;
-		{{- end }}
+			{{- if (entity.Persistence?.IsAudited ?? true) }}
+			DbContext.Entry(entity).State = EntityState.Deleted;
+			{{-else-}}
+			DbContext.{{entity.PluralName}}.Remove(entity);
+			{{- end}}
+		}
 
-		await OnCompletedAsync(request, entity);
+		await OnCompletedAsync(request, new {{entity.Name}}Entity());
```

Comment:
OnCompletedAsync accepts TRequest and TEntity. What should be sent in this case if we changed List<TEntity>?